apprise/plugins/NotifyDapnet.py

```diff
@@ -0,0 +1,393 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 Chris Caron <lead2gold@gmail.com>
+# All rights reserved.
+#
+# This code is licensed under the MIT License.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files(the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions :
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+# To use this plugin, sign up with Hampager (you need to be a licensed
+# ham radio operator
+#  http://www.hampager.de/
+#
+# You're done at this point, you only need to know your user/pass that
+# you signed up with.
+
+#  The following URLs would be accepted by Apprise:
+#   - dapnet://{user}:{password}@{callsign}
+#   - dapnet://{user}:{password}@{callsign1}/{callsign2}
+
+# Optional parameters:
+#   - priority (NORMAL or EMERGENCY). Default: NORMAL
+#   - txgroups --> comma-separated list of DAPNET transmitter
+#                           groups. Default: 'dl-all'
+#                           https://hampager.de/#/transmitters/groups
+
+from json import dumps
+
+# The API reference used to build this plugin was documented here:
+#  https://hampager.de/dokuwiki/doku.php#dapnet_api
+#
+import requests
+from requests.auth import HTTPBasicAuth
+
+from .NotifyBase import NotifyBase
+from ..AppriseLocale import gettext_lazy as _
+from ..URLBase import PrivacyMode
+from ..common import NotifyType
+from ..utils import is_call_sign
+from ..utils import parse_call_sign
+from ..utils import parse_list
+from ..utils import parse_bool
+
+
+class DapnetPriority(object):
+    NORMAL = 0
+    EMERGENCY = 1
+
+
+DAPNET_PRIORITIES = (
+    DapnetPriority.NORMAL,
+    DapnetPriority.EMERGENCY,
+)
+
+
+class NotifyDapnet(NotifyBase):
+    """
+    A wrapper for DAPNET / Hampager Notifications
+    """
+
+    # The default descriptive name associated with the Notification
+    service_name = 'Dapnet'
+
+    # The services URL
+    service_url = 'https://hampager.de/'
```

Comment:
I wonder if the protocol should change from `dapnet://` to `hampager://`. Or maybe additionally support the `hampager://` scheme as well?  Thoughts?

Reply:
That issue is indeed a tad confusing. 

Personally, I'd stick with the DAPNET name, though. Reason: In the end, "hampager" is more or less a reference to the target audience that is permitted to use this API and pager devices (ham radio operators and pager devices). DAPNET is however the  actual protocol that is used for the communication (decentralized amateur paging network).  And yes, both are being hosted by the same team.

But in the end, it's a question similar to e.g. using the Matrix messenger protocol with a third-party messenger client that is not provided with by the Matrix team itself. In this case, you'd also stick to e.g. ``matrix://`` and not the client software's name.

You can of course support both ``dapnet://`` and ``hampager://`` qualifiers - I don't see any issue with that. It might also make sense to capitalize the ``service_name`` variable's value but I'll leave that up to you.